status-page.py

```diff
@@ -1,72 +1,133 @@
-        }
-        self.incidents = {
-            'investigating': [],
-            'identified': [],
-            'watching': [],
-            'fixed': []
-        }
-        self.passed_tests = []
-        self.failed_tests = []
-
-    def parseTestResults(self):
-        self.passed_tests = []
-        self.failed_tests = []
+        self.broken_components = []
+        self.open_incidents = []
 
-    def getAllStatuses(self):
+    def get_broken_components(self):
         response = requests.get(f'{self.base_url}/components', self.header)
-        print(response)
         for component in response.json()['data']:
             status = component['status']
-            if status == 1:
-                self.components['operational'].append(component)
-            elif status == 2:
-                self.components['performance_issues'].append(component)
-            elif status == 3:
-                self.components['partial_outage'].append(component)
-            elif status == 4:
-                self.components['major_outage'].append(component)
-
-    def getAllIncidents(self):
+            if status != 1:
+                self.broken_components.append(component)
+
+    def get_all_open_incidents(self):
         response = requests.get(f'{self.base_url}/incidents', self.header)
-        print(response)
         for incident in response.json()['data']:
             status = incident['status']
-            if status == 1:
-                self.incidents['investigating'].append(incident)
-            if status == 2:
-                self.incidents['identified'].append(incident)
-            if status == 3:
-                self.incidents['watching'].append(incident)
-            if status == 4:
-                self.incidents['fixed'].append(incident)
+            if status != 4:
+                self.open_incidents.append(incident)
+
+    def update_incident_status(self):
+        # map api version and name to Cachet component IDs
+        component_map = {
+            'v1/academic-disciplines': 3,
+            'v1/advisors': 4,
+            'v1/xeapps': 5,
+            'v1/beaverbus': 6,
+            'v1/directory': 7,
+            'v2/directory': 8,
+            'v1/finance': 9,
+            'v1/hr': 10,
+            'v1/identify': 11,
+            'v1/locations': 12,
+            'v1/oauth2': 13,
+            'v1/onbase': 14,
+            'v1/persons': 15,
+            'v1/staff-fee-privilege': 16,
+            'v1/students': 17,
+            'v1/terms': 18,
+            'v1/textbooks': 19
+        }
+
+        # then, if there are any test failures, open an incident if there is none open already
+        for failed_test in self.log_json['failed_tests']:
+            already_reported = False
+            parsed_url = urlparse(failed_test['base_url'])
+            split_path = parsed_url.path.split('/')
+            api_version = split_path[1]
+            api_name = split_path[2]
+            version_and_name = f'{api_version}/{api_name}'
```

Comment:
Can't you just use regex to fetch this part in the first place?

Reply:
I'm thinking about leaving this as is because api_version and api_name are referenced in the request body, so they're useful variables to have. However, now that the the component map has been moved to the config, I could reformat the component map to a JSON object with version and name attributes, eliminating the need for string parsing

Reply:
I'm thinking about just leaving this as is. `api_version` and `api_name` are referenced in the request body, so they're useful variables. I guess using a regex could save that one call to urlparse(), but I'd say urlparse() is more readable than two regex's for the version and the name.